crates/astria-conductor/src/executor/queue.rs

```diff
@@ -0,0 +1,683 @@
+use std::collections::{
+    BTreeMap,
+    HashMap,
+};
+
+use color_eyre::eyre::Result;
+use tendermint::{
+    block::Height,
+    hash::Hash,
+};
+use tracing::info;
+
+use crate::types::SequencerBlockSubset;
+
+/// A queue for the SequencerBlockSubset type that holds blocks that are
+/// pending or not yet ready for execution.
+///
+/// This Queue handles all the fork choice logic for incoming Sequencer blocks
+/// from gossip. It is responsible for determining which blocks are safe to
+/// pass on to execution, and holds on to the other data until it is safe to
+/// execute, or deletes it if it is no longer needed or becomes stale or invalid.
+#[derive(Debug, Clone)]
+pub(super) struct Queue {
+    head_height: Height,
+    most_recent_soft_hash: Hash,
+
+    // The collection of all pending blocks. the blocks in this map at Height ==
+    // Queue.head_height are the head blocks
```

Comment:
head candidates is probably a more precise name to use through out the code. then we have that, a chain can have many head candidates and only one head, a head is chosen among head candidates.

Reply:
hm, I think I am porting this from somewhere else, let's stick to the terminology originating from geth terminology, `head` and `soft` for canonical head